src/layers/learning/channelwise_scale_bias.cpp

```diff
@@ -0,0 +1,136 @@
+  const auto local_bias = El::LockedView(local_weights,
+                                         El::ALL, El::IR(1));
+
+  // Dimensions
+  // Note: channel_size is the number of input entries per channel and
+  // local_width is the number of local mini-batch samples.
+  const auto dims = get_output_dims();
+  const El::Int num_channels = dims[0];
+  const El::Int channel_size = std::accumulate(dims.begin() + 1,
+                                               dims.end(),
+                                               1, std::multiplies<int>());
+  const El::Int local_width = local_input.Width();
+
+  // Apply channel-wise scale and bias
+  LBANN_OMP_PARALLEL_FOR
+  for (El::Int channel = 0; channel < num_channels; ++channel) {
+    const auto a = local_scale(channel, 0);
+    const auto b = local_bias(channel, 0);
+    const El::Int row_start = channel * channel_size;
+    const El::Int row_end = (channel + 1) * channel_size;
+    const El::Int col_start = 0;
+    const El::Int col_end = local_width;
+    for (El::Int col = col_start; col < col_end; ++col) {
+      for (El::Int row = row_start; row < row_end; ++row) {
+        const auto& x = local_input(row, col);
+        auto& y = local_output(row, col);
+        y = a * x + b;
+      }
+    }
+  }
+
+}
+
+template <>
+void channelwise_scale_bias_layer<data_layout::DATA_PARALLEL,El::Device::CPU>
+     ::bp_compute() {
+
+  // Local matrices
+  const auto& local_input = dynamic_cast<const CPUMat&>(get_local_prev_activations());
+  const auto& local_gradient_wrt_output = dynamic_cast<const CPUMat&>(get_local_prev_error_signals());
+  auto& local_gradient_wrt_input = dynamic_cast<CPUMat&>(get_local_error_signals());
+  const auto& local_weights = dynamic_cast<const CPUMat&>(m_weights[0]->get_values().LockedMatrix());
+  auto& local_gradient_wrt_weights = dynamic_cast<CPUMat&>(m_weights_gradient->Matrix());
+  const auto local_scale = El::LockedView(local_weights,
+                                          El::ALL, El::IR(0));
+  auto local_gradient_wrt_scale = El::View(local_gradient_wrt_weights,
+                                           El::ALL, El::IR(0));
+  auto local_gradient_wrt_bias = El::View(local_gradient_wrt_weights,
+                                          El::ALL, El::IR(1));
+
+
+  // Dimensions
+  // Note: channel_size is the number of input entries per channel and
+  // local_width is the number of local mini-batch samples.
+  const auto dims = get_output_dims();
+  const El::Int num_channels = dims[0];
+  const El::Int channel_size = std::accumulate(dims.begin() + 1,
+                                               dims.end(),
+                                               1, std::multiplies<int>());
+  const El::Int local_width = local_input.Width();
+
+  // Compute gradients
+  LBANN_OMP_PARALLEL_FOR
+  for (El::Int channel = 0; channel < num_channels; ++channel) {
+    const auto a = local_scale(channel, 0);
+    DataType da{0}, db{0};
```

Comment:
Why not accumulate directly into `local_gradient_wrt_scale`/`bias`?

Reply:
I don't expect there to be a significant difference.